### PR TITLE
Make ListView Context Menu show up in correct position on Windows

### LINKED ISSUE #6490: Context menu on wrong position in Windows
### Description

I have a ListView with ViewCell as base in the DataTemplate. The ViewCell has ContextActions.  In Windows the menu is shown, but not where mouse pointer were when the user right clicked.

[Demo project](https://github.com/pekspro/MauiIssues/tree/BadContextMenuPosition)

### Steps to Reproduce

1. Create a new MAUI project.
2. Add this in the main page:
``` XAML
    <ListView x:Name="ItemsList">
        <ListView.ItemTemplate>
            <DataTemplate>
                <ViewCell>
                    <ViewCell.ContextActions>
                        <MenuItem Text="Detail"></MenuItem>
                        <MenuItem Text="Delete"></MenuItem>
                    </ViewCell.ContextActions>
                    <Grid>
                        <Label Text="{Binding }" FontSize="25"></Label>
                    </Grid>
                </ViewCell>
            </DataTemplate>
        </ListView.ItemTemplate>
    </ListView>
```
3 Initialize the list in the constructor:
``` C#
	public MainPage()
	{
		InitializeComponent();

		ItemsList.ItemsSource = new string[] { "Hello", "World" }; 
	}
```

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

Windows 10.0.17763.0

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Eilon**
### Description of Change

Fixes the ListView to show the context menu where the mouse was right-clicked instead of in some seemingly-random location.

There are two parts to this fix:

1. Use the correct Win App SDK event to detect the right click (or on touch devices, a press&hold)
2. Tell the flyout menu where to show by using the aforementioned event's event args, which tell you where the click took place

Notes:
* Android already works: it shows a special "menu" at the top of the screen
* iOS already works: the ListView has a slidy/swipey thing it shows with the menu actions
* MacCatalyst completely doesn't work - the menus are not available anywhere/anyhow. I logged #8664 to track that.

### Issues Fixed

Fixes #6490

### CODE REVIEW

File: src/Controls/src/Core/Compatibility/Handlers/ListView/Windows/CellControl.cs [hunk 1]
```diff
@@ -320,7 +312,14 @@ void OpenContextMenu()
...
 			}
 
-			FlyoutBase.ShowAttachedFlyout(CellContent);
+			FlyoutBase
+				.GetAttachedFlyout(CellContent)
+				.ShowAt(
+					CellContent,
+					new FlyoutShowOptions
+					{
+						Position = point,
```
Comment (Eilon) [inline src/Controls/src/Core/Compatibility/Handlers/ListView/Windows/CellControl.cs:321]: This is the key part of the change: to actually show the menu where the user clicked/tapped. All the rest is plumbing changes.

File: src/Controls/src/Core/Compatibility/Handlers/ListView/Windows/CellControl.cs [hunk 2]
```diff
@@ -450,8 +448,7 @@ void SetupContextMenu()
 				return;
 			}
 
-			CellContent.PointerReleased += OnClick;
-			CellContent.Holding += OnLongTap;
+			CellContent.RightTapped += OnCellContentRightTapped;
```
Comment (Eilon) [inline src/Controls/src/Core/Compatibility/Handlers/ListView/Windows/CellControl.cs:451]: See this comment about why I changed to use this event: https://github.com/dotnet/maui/issues/6490#issuecomment-1180857070

TL;DR: The docs say it's the best event to use for this, and it also works better than the previous two events combined.

**Review by jsuarezruiz — APPROVED** [review]

**Comment by Eilon** [comment]
Thank you for the review @jsuarezruiz !
